wasm/wazeroir/interpreter.go

```diff
@@ -0,0 +1,1502 @@
+		case OperationKindLoad32:
+			{
+				base := op.us[1] + it.pop()
+				if op.b1 == 1 {
+					it.push(uint64(int32(binary.LittleEndian.Uint32(memoryInst.Buffer[base:]))))
+				} else {
+					it.push(uint64(binary.LittleEndian.Uint32(memoryInst.Buffer[base:])))
+				}
+				frame.pc++
+			}
+		case OperationKindStore:
+			{
+				val := it.pop()
+				base := op.us[1] + it.pop()
+				switch SignLessType(op.b1) {
+				case SignLessTypeI32, SignLessTypeF32:
+					binary.LittleEndian.PutUint32(memoryInst.Buffer[base:], uint32(val))
+				case SignLessTypeI64, SignLessTypeF64:
+					binary.LittleEndian.PutUint64(memoryInst.Buffer[base:], val)
+				}
+				frame.pc++
+			}
+		case OperationKindStore8:
+			{
+				val := byte(it.pop())
+				base := op.us[1] + it.pop()
+				memoryInst.Buffer[base] = val
+				frame.pc++
+			}
+		case OperationKindStore16:
+			{
+				val := uint16(it.pop())
+				base := op.us[1] + it.pop()
+				binary.LittleEndian.PutUint16(memoryInst.Buffer[base:], val)
+				frame.pc++
+			}
+		case OperationKindStore32:
+			{
+				val := uint32(it.pop())
+				base := op.us[1] + it.pop()
+				binary.LittleEndian.PutUint32(memoryInst.Buffer[base:], val)
+				frame.pc++
+			}
+		case OperationKindMemorySize:
+			{
+				v := uint64(int32(uint64(len(frame.f.moduleInstance.Memory.Buffer)) / wasm.PageSize))
+				it.push(v)
+				frame.pc++
+			}
+		case OperationKindMemoryGrow:
+			{
+				n := it.pop()
+				max := uint64(math.MaxUint32)
+				if memoryInst.Max != nil {
+					max = uint64(*memoryInst.Max) * wasm.PageSize
+				}
+				if uint64(n*wasm.PageSize+uint64(len(memoryInst.Buffer))) > max {
+					v := int32(-1)
+					it.push(uint64(v))
+				} else {
+					it.push(uint64(len(memoryInst.Buffer)) / wasm.PageSize)
+					memoryInst.Buffer = append(memoryInst.Buffer, make([]byte, n*wasm.PageSize)...)
+				}
+				frame.pc++
+			}
+		case OperationKindConstI32, OperationKindConstI64,
+			OperationKindConstF32, OperationKindConstF64:
+			{
+				it.push(op.us[0])
+				frame.pc++
+			}
+		case OperationKindEq:
+			{
+				var b bool
+				switch SignLessType(op.b1) {
+				case SignLessTypeI32, SignLessTypeI64:
+					v2, v1 := it.pop(), it.pop()
+					b = v1 == v2
+				case SignLessTypeF32:
+					v2, v1 := it.pop(), it.pop()
+					b = math.Float32frombits(uint32(v2)) == math.Float32frombits(uint32(v1))
+				case SignLessTypeF64:
+					v2, v1 := it.pop(), it.pop()
+					b = math.Float64frombits(v2) == math.Float64frombits(v1)
+				}
+				if b {
+					it.push(1)
+				} else {
+					it.push(0)
+				}
+				frame.pc++
+			}
+		case OperationKindNe:
+			{
+				var b bool
+				switch SignLessType(op.b1) {
+				case SignLessTypeI32, SignLessTypeI64:
+					v2, v1 := it.pop(), it.pop()
+					b = v1 != v2
+				case SignLessTypeF32:
+					v2, v1 := it.pop(), it.pop()
+					b = math.Float32frombits(uint32(v2)) != math.Float32frombits(uint32(v1))
+				case SignLessTypeF64:
+					v2, v1 := it.pop(), it.pop()
+					b = math.Float64frombits(v2) != math.Float64frombits(v1)
```

Comment:
fromBits here and elsewhere

Reply:
the bit size is different from the above :D

Reply:
nm that's a case format inconsistency in the stdlib!